Core/DISET/AuthManager.py

```diff
@@ -256,6 +262,26 @@ def getUsername(self, credDict):
       return True
     return False
 
+  def isUserSuspended(self, credDict):
+    """ Discover if the use is in Suspended status
+
+    :param credDict: Credentials to ckeck
+    :return: Boolean True if user is Suspended
+    """
+    if self.KW_USERNAME not in credDict:
+      self.getUsername(credDict)
```

Comment:
Looking at it, I don't really understand what `getUsername` does. According to the docstring it is supposed to return a bool (that is not checked btw) if the username could be resolved. And the first thing it does is return `True` if there is no DN in the dict...

Reply:
it updates credDict

Reply:
But isn't that a bug ? https://github.com/DIRACGrid/DIRAC/blob/2cbd221beb4c3613bc7b080af54344375f265759/Core/DISET/AuthManager.py#L250-L251